### ISSUE TITLE
restic init in existing bucket

### ISSUE THREAD

**Issue by nunoperalta** [restic/restic#2404]
Output of restic version
--------------------------

restic 0.9.5 compiled with go1.12.4 on linux/amd64

How did you run restic exactly?
-------------------------------

export AWS_ACCESS_KEY_ID=
export AWS_SECRET_ACCESS_KEY=
export RESTIC_PASSWORD=
export RESTIC_REPOSITORY=
restic init

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

CentOS (using Ansible)

Expected behavior
-----------------

Creates initial structure without failing because bucket exists.

Actual behavior
---------------

"Fatal: create repository at s3:https://s3.amazonaws.com/xxxxxxx failed: client.BucketExists: Head https://xxxxxxx.s3.dualstack.region-1.amazonaws.com/: 301 response missing Location header"

Steps to reproduce the behavior
-------------------------------

Create S3 Bucket,
add encryption,
add versioning,
set up lifecycle,
etc...
(prepare bucket as you want)

Keep bucket 100% empty.

Run restic init.

Do you have any idea what may have caused this?
-----------------------------------------------

Bucket already exists, restic doesn't seem to like that.

Do you have an idea how to solve the issue?
-------------------------------------------

Allow us to init a repository on an empty bucket.

**Comment by jtagcat** [comment]
Besides searching for issues first, I'd recommend commenting on your issue:
"Closed as duplicate of #0000" and closing it. If it is not a duplicate, but just a 'dumb' question, answer and close the issue!
This way others searching for the issue in the same way you did, with similar thinking procedure are able to benefit from this.

**Comment by nunoperalta** [comment]
I realized that the problem isn't that "BucketExists". That's just informational, apparently.

I see that giving the URL: https://s3.amazonaws.com/BUCKETNAME

Makes Restic assume it is on "us-east-1" region, but the bucket is in "us-west-2"...

So, giving the URL instead: https://s3-us-west-2.amazonaws.com/BUCKETNAME
Resolved the problem.

There's absolutely NOTHING in the error that says that...
The error says "create repository failed" and then right after says "BucketExists",
so the error is completely misleading.

But, seeing the "301 response missing Location header" eventually made me think that the error could be something else, and find the problem.
